src/printer.js

```diff
@@ -621,7 +621,40 @@ function genericPrintNoParens(path, options, print) {
         } else {
           parts.push(printPropertyKey(path, print));
         }
-        parts.push(": ", path.call(print, "value"));
+        let noBreak = [
+          "ArrayExpression",
+          "ArrayPattern",
+          "ArrowFunctionExpression",
+          "AssignmentPattern",
+          "CallExpression",
+          "FunctionExpression",
+          "Identifier",
+          "Literal",
+          "LogicalExpression",
+          "MemberExpression",
+          "NewExpression",
+          "ObjectExpression",
+          "ObjectPattern",
+          "StringLiteral",
+          "ThisExpression",
+          "TypeCastExpression"
+        ].reduce((stack, type) => stack || (n.value.type === type), false);
```

Comment:
I think is valuable info. There might be other places we could use this logic as well. Can you move it out to a utility function at the bottom of the file? Something like `shouldPrintSameLine` or something. Not sure what the best name is.

This function just takes a node and checks these. We can also use `namedTypes.<type>.check` to check a type of node, but looking at the code I'm worried it's a bit slow, and we already don't use it many other places, so I dunno... we should at least use it for `Literal` because it works for both babylon and flow even though they define the literal types differently.  Although it looks inconsistent, the function would look like:

```js
function shouldPrintSameLine(node) {
  const type = node.type;
  return namedTypes.Literal.check(node) ||
    type === "ArrayExpression" ||
    type === "ArrayPattern" ||
    type === "ArrowFunctionExpression" ||
    ...
}
```

I'd like to avoid the `reduce` call for performance too, even if the above is a little more verbose.